compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -475,6 +476,28 @@ convertSimpleRecordDef env tycon = do
 
     pure $ typeDef : [workerDef | flavour == NewtypeFlavour]
 
+convertTypeSynonym :: Env -> TyCon -> ConvertM [Definition]
+convertTypeSynonym env tycon
+    | NameIn DA_Generics _ <- GHC.tyConName tycon
+    = pure []
+
+    | envLfVersion env `supports` featureTypeSynonyms
+    , Just (params, body) <- synTyConDefn_maybe tycon
+    , isLiftedTypeKind (tyConResKind tycon)
```

Comment:
Does that check that we only export synonyms of kind star?

Reply:
So ... I just tested and this does indeed export constraint synonyms as well. But it also exports them without problem in data-dependencies (they work correctly, according to my tests).

So is this better to reject constraint synonyms (making the implementation more complicated), or to add a test to make sure they work (without changing the implementation)?

Reply:
(cc @cocreature ^^)

Reply:
I’d say support it and add a test for it.

Reply:
Supporting and testing it sounds fine.